### PR TITLE
#1713 Create ReportChart component

### LINKED ISSUE #1713: ReportChart component
## Is your feature request related to a problem? Please describe.

Need a chart component for displaying charts/graphs for the mobile dashboard 

## Describe the solution you'd like

A component similar to https://github.com/openmsupply/mobile/blob/49d8ab764c85ec5e35699661da92aad8c3e1b03a/src/widgets/ReportChart.js

However, I *think* a better implementation would be to separate out the different charts - pie/line/bar into their own components, rather than having a single component.

## Implementation

As above. Whoever the implementer is can decide or we can discuss in the issue whether individual components or a single component is best.

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by diegoevangelisti**
Fixes #1713

## Change summary

This component manages which is the report that is needed to be render (PieChart, BarChart, LineChart, TableReport) according to its `Type`.

## Testing

Further testing will be taken once the feature/dashboard is intended to be merged into develop.

### Related areas to think about

Codebase taken from past work: [branch](https://github.com/openmsupply/mobile/blob/49d8ab764c85ec5e35699661da92aad8c3e1b03a/src/widgets/ReportChart.js). Further changes may be taken into consideration.

### CODE REVIEW

File: src/widgets/ReportChart.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  // Victory Native sizes are set using absolute values. Parents dimensions are used to
+  // calculate relative values for width and height for each chart.
+  const onLayout = event => {
+    const newDimensionsObj = {
+      height: event.nativeEvent.layout.width,
+      width: event.nativeEvent.layout.height,
+    };
+    setDimensions(newDimensionsObj);
+  };
```
Comment (joshxg) [inline src/widgets/ReportChart.js]: I think if we're doing this here we should *only* do it here, not in the children. So, that would mean passing to the charts i.e.

```
const BarChart = ({width, height, report}) => {
 ....
}
```

and using that here :

```
case 'BarChart':
    return <BarChart report={report} width={width} height={height}/>
```

There doesn't seem to be a reason to do it twice

Alternatively, you can just keep it in the chart components, and have them do the check for width/height in all of them, but I think it's nicer to do it the one time in here

Comment (diegoevangelisti) [reply src/widgets/ReportChart.js]: I agree! that's what we discussed yesterday afternoon and totally happy with keeping that in the parent and take it out from children components.

Comment (diegoevangelisti) [reply src/widgets/ReportChart.js]: I will create a issue to remove width and height calculation on children components as well.

Comment (joshxg) [reply src/widgets/ReportChart.js]: ok sure!

**Pull request merged**